tests/dual-farming/single-reward.ts

```diff
@@ -0,0 +1,219 @@
+        stakingMint,
+        stakingVault: stakingVaultAddress,
+        systemProgram: anchor.web3.SystemProgram.programId,
+        tokenProgram: TOKEN_PROGRAM_ID,
+      })
+      .signers([BASE_KEYPAIR, ADMIN_KEYPAIR])
+      .rpc();
+  });
+
+  it("should create new user", async () => {
+    const [farmingPoolAddress, _farmingPoolBump] = await getPoolPda(
+      program, stakingMint, BASE_KEYPAIR.publicKey
+    );
+    const [userStakingAddress, _userStakingAddressBump] = await getUserPda(
+      program,
+      farmingPoolAddress,
+      USER_KEYPAIR.publicKey
+    );
+
+    await program.methods
+      .createUser()
+      .accounts({
+        owner: USER_KEYPAIR.publicKey,
+        pool: farmingPoolAddress,
+        systemProgram: anchor.web3.SystemProgram.programId,
+        user: userStakingAddress,
+      })
+      .signers([USER_KEYPAIR])
+      .rpc();
+  });
+  it("should stake to the pool", async () => {
+    const STAKE_AMOUNT = new anchor.BN(500 * TOKEN_MULTIPLIER);
+
+    await stakingToken.mintTo(
+      userStakingATA,
+      ADMIN_KEYPAIR,
+      [],
+      1000 * TOKEN_MULTIPLIER
+    );
+
+    const [farmingPoolAddress, _farmingPoolBump] = await getPoolPda(
+      program, stakingMint, BASE_KEYPAIR.publicKey
+    );
+
+    const [userStakingAddress, _userStakingAddressBump] = await getUserPda(
+      program,
+      farmingPoolAddress,
+      USER_KEYPAIR.publicKey
+    );
+
+    const poolAccount = await program.account.pool.fetch(farmingPoolAddress);
+
+    await program.methods
+      .stake(STAKE_AMOUNT)
+      .accounts({
+        owner: USER_KEYPAIR.publicKey,
+        pool: farmingPoolAddress,
+        stakeFromAccount: userStakingATA,
+        stakingVault: poolAccount.stakingVault,
+        tokenProgram: TOKEN_PROGRAM_ID,
+        user: userStakingAddress,
+      })
+      .signers([USER_KEYPAIR])
+      .rpc();
+  });
+
+  it("should fund the pool reward A only", async () => {
+    const FUND_AMOUNT = new anchor.BN(20_000 * TOKEN_MULTIPLIER);
+
+    const [farmingPoolAddress, _farmingPoolBump] = await getPoolPda(
+      program, stakingMint, BASE_KEYPAIR.publicKey
+    );
+
+    const poolAccount = await program.account.pool.fetch(farmingPoolAddress);
+
+    await rewardToken.mintTo(
+      adminRewardATA,
+      ADMIN_KEYPAIR,
+      [],
+      100_000 * TOKEN_MULTIPLIER
+    );
+
+
+    await program.methods
+      .fund(FUND_AMOUNT, 0)
+      .accounts({
+        fromA: adminRewardATA,
+        fromB: adminRewardATA,
+        funder: ADMIN_KEYPAIR.publicKey,
+        pool: farmingPoolAddress,
+        rewardAVault: poolAccount.rewardAVault,
+        rewardBVault: poolAccount.rewardBVault,
+        stakingVault: poolAccount.stakingVault,
+        tokenProgram: TOKEN_PROGRAM_ID,
+      })
+      .signers([ADMIN_KEYPAIR])
+      .rpc()
+      .catch(console.log);
+    let rewardBalance = await provider.connection.getTokenAccountBalance(userRewardATA);
+    console.log(rewardBalance);
```

Comment:
Should we use assert instead of console.log?

Reply:
Phi had fixed it

Reply:
Phi had fixed it